spec/webdrivers/chromedriver_spec.rb

```diff
@@ -5,125 +5,207 @@
 describe Webdrivers::Chromedriver do
   let(:chromedriver) { described_class }
 
-  it 'updates' do
-    chromedriver.update
+  before do
+    chromedriver.remove
+    chromedriver.version = nil
   end
 
-  it 'parses chromedriver versions before 2.10' do
-    expect(chromedriver.send(:normalize_version, '2.9').version).to eq '2.9'
-  end
+  describe '#update' do
+    context 'when evaluating #correct_binary?' do
+      it 'does not download when latest version and current version match' do
+        allow(chromedriver).to receive(:latest_version).and_return(Gem::Version.new('72.0.0'))
+        allow(chromedriver).to receive(:current_version).and_return(Gem::Version.new('72.0.0'))
 
-  it 'finds latest version' do
-    old_version = Gem::Version.new('2.30')
-    future_version = Gem::Version.new('80.00')
-    latest_version = chromedriver.latest_version
+        chromedriver.update
 
-    expect(latest_version).to be > old_version
-    expect(latest_version).to be < future_version
-  end
+        expect(chromedriver.send(:downloaded?)).to be false
+      end
 
-  it 'downloads latest release for current version of Chrome by default' do
-    chromedriver.remove
-    chromedriver.download
-    cur_ver    = chromedriver.current_version.version
-    latest_ver = chromedriver.latest_version.version
-    expect(cur_ver).to eq latest_ver
-  end
+      it 'does not download when offline, binary exists and is less than v70' do
+        allow(Net::HTTP).to receive(:get_response).and_raise(SocketError)
+        allow(chromedriver).to receive(:downloaded?).and_return(true)
```

Comment:
Not 100% sure but wouldn't this stop it from being downloaded - and not actually test that it wasn't downloaded

Reply:
This is testing if it is offline, and a binary is there, it doesn't still download.
setting `#downloaded?` to true is mocking that something is there. Then if the file doesn't actually exist, then we know it wasn't still downloaded. It's just stepping through the conditional in the method to make sure it short-circuits.